### PR TITLE
feat: announce: slack

### LINKED ISSUE #2425: add Slack as a new GoReleaser Announce #2424
Is your feature request related to a problem? Please describe.
It is not related to a problem, it is a new feature.

Describe the solution you'd like
We (w/@Dentrax) would like to add another integration (Slack in this case) to the GoReleaser Announce feature.

Slack also has Golang SDK for its API  which you can see from [here](https://github.com/nlopes/slack). So, we can easily add this support to the GoReleaser Announce feature.

### PR DESCRIPTION
**Pull request by Dentrax**
This PR adds support for Slack Announce!

Fixes #2425

Signed-off-by: Furkan <furkan.turkal@trendyol.com>

### CODE REVIEW

**Comment by Dentrax** [comment]
> I think so, yes. We can keep on env only the "secrets".

Updated. ✌️

File: www/docs/customization/announce.md [hunk 1]
```diff
@@ -30,5 +30,30 @@ announce:
...
+  slack:
+    # Wether its enabled or not.
+    # Defaults to false.
+    enabled: true
+
+    # Message template to use while publishing.
+    # Defaults to `{{ .ProjectName }} {{ .Tag }} is out! Check it out at {{ .GitURL }}/releases/tag/{{ .Tag }}`
+    message_template: 'Awesome project {{.Tag}} is out!'
+```
+
```
Comment (caarlos0) [inline www/docs/customization/announce.md:57]: docs are outdated now, aren't they? we changed most props to be on the config file instead of yaml...

Comment (Dentrax) [reply www/docs/customization/announce.md]: Oh, I forgot to update docs. Updated. 👍

**Comment by caarlos0** [comment]
there are some conflicts, can you merge with master? other than that, lgtm 🙏

**Comment by Dentrax** [comment]
Screen Shot 2021-09-01 at 21 47 00

@developer-guy 🤩

**Comment by Dentrax** [comment]
> there are some conflicts, can you merge with master? other than that, lgtm 🙏

I think I fixed the conflicts. Can you please review again? 🤗 

_We tested this new announce on our upcoming private project [kink](https://github.com/Trendyol/kink) (kind in kubernetes), it will out soon._

File: www/docs/customization/announce.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 title: Announce
 ---
 
-GoReleaser can also announce new releases, currently, to Twitter only.
+GoReleaser can also announce new releases, currently, to Twitter and Slack only.
```
Comment (caarlos0) [inline www/docs/customization/announce.md:5]:
```suggestion
GoReleaser can also announce new releases, currently, to Twitter, Reddit and Slack only.
```

File: www/docs/customization/announce.md [hunk 3]
```diff
@@ -30,6 +30,39 @@ announce:
...
+    channel: '#channel'
+
+    # Set your Webhook's user name.
+    username: ''
+
+    # Emoji to use as the icon for this message. Overrides icon_url.
+    icon_emoji: ''
+
+    # URL to an image to use as the icon for this message.
+    icon_url: ''
```
Comment (caarlos0) [inline www/docs/customization/announce.md:63]: maybe this could default to `https://github.com/goreleaser.png` 🤔

Comment (caarlos0) [reply www/docs/customization/announce.md:63]: actually, nevermind

**Review by caarlos0 — APPROVED** [review]

**Comment by caarlos0** [comment]
Will merge and fix the conflict manually

**Comment by caarlos0** [comment]
thanks again for the hard work on this 💛

File: www/docs/customization/announce.md (hunk 3 above)
Comment (Dentrax) [reply www/docs/customization/announce.md:63]: This icon would be great! +1 for this idea.

**Pull request merged**